store/root/store.go

```diff
@@ -54,25 +54,31 @@ type Store struct {
 
 	// traceContext defines the tracing context, if any, for trace operations
 	traceContext store.TraceContext
+
+	// pruningManager manages pruning of the SS and SC backends
+	pruningManager *pruning.Manager
 }
 
 func New(
 	logger log.Logger,
 	initVersion uint64,
 	ss store.VersionedDatabase,
-	sc *commitment.Database,
+	sc store.Committer,
 ) (store.RootStore, error) {
 	rootKVStore, err := branch.New(defaultStoreKey, ss)
 	if err != nil {
 		return nil, err
 	}
 
+	pruningManager := pruning.NewManager(ss, sc, logger)
```

Comment:
How is the lifecycle managed? If it's created here, where do we start and stop it?